### ISSUE TITLE
[BUG] Can't load namespace creation page content and create namespace on Harvester from Rancher v2.8.5

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#6953]
**Describe the bug**

When we try to create a namespace from the Rancher virtualization management.

* Can't load the namespace creation page content well 

  ![image](https://github.com/user-attachments/assets/78715daa-3156-49a1-af77-2e64549a8e9f)

* When given the name and click create, it would prompt failure message

  ```
  Failed to parse body: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type map[string]interface {}
  ```

  ![image](https://github.com/user-attachments/assets/6abdd271-c61e-4621-8d97-01f59992e592)

  https://github.com/user-attachments/assets/8d4540bd-dd63-4c73-82f1-59bb47eff6b7

**To Reproduce**

  1. Import Harvester to Rancher
  2. Access Harvester from Rancher virtualization management 
  3. Go to `Projects/Namespaces` and create a new project `test-project`
  4. In the Resource Quotas tab, add Storage Reservation and use 20000 MiB as project limit and 12000 MiB as namespace default limit
    ![image](https://github.com/user-attachments/assets/b0a52f0e-06a7-4700-879c-bbbd09638bbc)
  
  5. Create a new namespace named `quota-test` in the previously created project. 

**Expected behavior**

Can load namespace creation page well on Harvester from Rancher virtualization management. 

**Support bundle**

[supportbundle_c2f1e2b2-34c5-4cbb-87e2-560ff5dd2a88_2024-11-05T11-19-29Z.zip](https://github.com/user-attachments/files/17631628/supportbundle_c2f1e2b2-34c5-4cbb-87e2-560ff5dd2a88_2024-11-05T11-19-29Z.zip)

**Environment**
 - Harvester ISO version: `v1.4-eaf3009a-head` (24/11/05)
 - Underlying Infrastructure: Three nodes kvm machines 

**Additional context**
This issue did not happen on Rancher v2.9.2

**Comment by TachunLin** [comment]
This issue also exists on Rancher `v2.8.9` with Harvester `v1.4-eaf3009a-head`
![image](https://github.com/user-attachments/assets/ccb5ccdf-93db-4245-be9c-94a6e10f1a22)

**Comment by a110605** [comment]
Due to https://github.com/harvester/dashboard/pull/1208 added steve-schema.ts in v1.4.0-rc5 caused below error from Rancher shell v2.8.x [shell/utils/create-yaml.ts](https://github.com/rancher/dashboard/blob/a38998e501e4ffa29399414f50633569aff3a60b/shell/utils/create-yaml.js#L130-L133)

Test with v1.4.0-rc4  + Rancher-2.8.5 is good. cc @torchiaf 

```
Error: Cannot find resourceFields for Schema namespace (schemaDefinitions have not been fetched) 
```

<img width="1494" alt="Screenshot 2024-11-07 at 10 40 30 AM" src="https://github.com/user-attachments/assets/0670cca4-a455-42d4-a0ab-2a7b7327ddf0">

**Comment by a110605** [comment]
[Rancher v2.8](https://github.com/rancher/dashboard/tree/v2.8.6/shell/models) -> [Rancher v2.9](https://github.com/rancher/dashboard/tree/v2.9.0/shell/models) had lots of code change (even broken change), like **steve-schema.ts** was introduced in v2.9, but not existed on v2.8.

This diff somehow may break something on v2.8.x. 

See @torchiaf has any solution or workaround

BTW, does Harvester v1.4.0 needs to support both Rancher v2.8.x and v2.9.x ? cc @bk201

**Comment by bk201** [comment]
yes, QA validates the release with Rancher 2.8.9.

**Comment by TachunLin** [comment]
Verified fixed on `v1.4-eaf3009a-head` (24/11/08). Close this issue.

### Test result of Rancher v2.8.9
<details><summary>$\color{green}{\textsf{PASS}}$ Can load namespace creation page content and create namespace well $~~$</summary>

  ![image](https://github.com/user-attachments/assets/150cb722-8b5f-4061-91b4-9e0b08c97047)
  
  ![image](https://github.com/user-attachments/assets/88b3b36a-1c3c-407a-a56f-d6e75e221daf)


 </details>


### Test result of Rancher v2.9.3
<details><summary>$\color{green}{\textsf{PASS}}$ Can load namespace creation page content and create namespace well $~~$</summary>

  ![image](https://github.com/user-attachments/assets/392d9aa2-78ad-4e5e-9a6c-9e63e6635a34)
  
  ![image](https://github.com/user-attachments/assets/55d167b6-d36a-44f3-ad9d-6ce0772ce05e)


 </details>

### Test Information
* Test Environment: Single node kvm machine
* Harvester version: `v1.4-eaf3009a-head` (24/11/08)
* Rancher: v2.8.9, v2.9.3

### Verify Steps
<details><summary> Create namespace in Rancher </summary>

1. Import Harvester to Rancher
2. Access Harvester from Rancher virtualization management
3. Go to Projects/Namespaces and create a new project test-project
4. In the Resource Quotas tab, add Storage Reservation and use 20000 MiB as project limit and 12000 MiB as namespace default limit
5. Create a new namespace named quota-test in the previously created project.

</details>
